From 32c391ffa6167c77ba5db190a60d78d23331355e Mon Sep 17 00:00:00 2001
From: Gao Zhi Gang <zggao@suse.com>
Date: Thu, 13 Jul 2017 12:22:37 -0400
Subject: [PATCH] used in yocto

---
 Makefile | 13 ++++++-------
 file.c   |  9 +++++++++
 2 files changed, 15 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index 3edb4e2..dcec3d8 100644
--- a/Makefile
+++ b/Makefile
@@ -13,21 +13,20 @@ ccflags-$(CONFIG_NTFS_DEBUG)	+= -DDEBUG
 ccflags-$(CONFIG_NTFS_RW)	+= -DNTFS_RW
 else
 
-KERNEL ?= /lib/modules/`uname -r`/build
+KERNEL_SRC ?= /lib/modules/`uname -r`/build
 
 default:
-	CONFIG_NTFS_FS=m CONFIG_NTFS_RW=y CONFIG_NTFS_DEBUG=y $(MAKE) -C $(KERNEL) M=$$PWD
+	CONFIG_NTFS_FS=m CONFIG_NTFS_RW=y CONFIG_NTFS_DEBUG=y $(MAKE) -C $(KERNEL_SRC) M=$$PWD
 
 
 .PHONY : install help clean
 help:
-	$(MAKE) -C $(KERNEL) M=$$PWD help
+	$(MAKE) -C $(KERNEL_SRC) M=$$PWD help
 
-install : default
-	$(MAKE) -C $(KERNEL) M=$$PWD modules_install
-	depmod -A
+modules_install : default
+	$(MAKE) -C $(KERNEL_SRC) M=$$PWD modules_install
 
 clean:
-	make -C $(KERNEL) M=`pwd` clean
+	make -C $(KERNEL_SRC) M=`pwd` clean
 
 endif
diff --git a/file.c b/file.c
index 1475243..7b3c8a4 100644
--- a/file.c
+++ b/file.c
@@ -746,7 +746,12 @@ map_buffer_cached:
 					set_buffer_uptodate(bh);
 				if (unlikely(was_hole)) {
 					/* We allocated the buffer. */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,12,0)
 					clean_bdev_bh_alias(bh);
+#else
+					unmap_underlying_metadata(bh->b_bdev,
+							bh->b_blocknr);
+#endif
 					if (bh_end <= pos || bh_pos >= end)
 						mark_buffer_dirty(bh);
 					else
@@ -789,7 +794,11 @@ map_buffer_cached:
 				continue;
 			}
 			/* We allocated the buffer. */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,12,0)
 			clean_bdev_bh_alias(bh);
+#else
+			unmap_underlying_metadata(bh->b_bdev, bh->b_blocknr);
+#endif
 			/*
 			 * If the buffer is fully outside the write, zero it,
 			 * set it uptodate, and mark it dirty so it gets
-- 
2.12.0

