From 222094498afeea023542ab3193c328e7c80168b7 Mon Sep 17 00:00:00 2001
From: Gao Zhi Gang <zggao@suse.com>
Date: Thu, 8 Jun 2017 10:55:29 -0400
Subject: [PATCH]  only git commit time/localtime.c

---
 time/localtime.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/time/localtime.c b/time/localtime.c
index 07dd67ca71..17c2e528fb 100644
--- a/time/localtime.c
+++ b/time/localtime.c
@@ -30,12 +30,19 @@ __localtime_r (const time_t *t, struct tm *tp)
   return __tz_convert (t, 1, tp);
 }
 weak_alias (__localtime_r, localtime_r)
+#include <sys/types.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <signal.h>
+#include <stdio.h>
 
 
 /* Return the `struct tm' representation of *T in local time.  */
 struct tm *
 localtime (const time_t *t)
 {
+    printf("tid[%ld]\n",  syscall(__NR_gettid));
   return __tz_convert (t, 1, &_tmbuf);
 }
 libc_hidden_def (localtime)
-- 
2.12.0

